Modules/Devices/src/Devices.Application/Identities/Commands/DeletionProcessGracePeriod/Handler.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Backbone.BuildingBlocks.Application.PushNotifications;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.Infrastructure.PushNotifications.DeletionProcess;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using Backbone.Tooling;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Identities.Commands.DeletionProcessGracePeriod;
+public class Handler : IRequestHandler<DeletionProcessGracePeriodCommand, DeletionProcessGracePeriodResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly IPushNotificationSender _pushSenderService;
+
+    public Handler(IIdentitiesRepository identitiesRepository, IPushNotificationSender pushSenderService)
+    {
+        _identitiesRepository = identitiesRepository;
+        _pushSenderService = pushSenderService;
+    }
+
+    public async Task<DeletionProcessGracePeriodResponse> Handle(DeletionProcessGracePeriodCommand request, CancellationToken cancellationToken)
+    {
+        var identities = await _identitiesRepository.FindAllWithApprovedDeletionProcess(cancellationToken, track: true);
+
+        foreach (var identity in identities)
+        {
+            var deletionProcess = identity.DeletionProcesses.First(d => d.IsActive());
+
+            if (deletionProcess.GracePeriodReminder1SentAt == null)
```

Comment:
Shouldn't we validate which reminders we are going to send?
Otherwise we might end up sending 3 reminders at once (obviously only some specific scenarios e.g., configuration changes)

Reply:
Is it expected to only be able to send 1 notification per run of the handler for a given process?
So, in a scenario where for some reason we couldn't deliver the first two notifications in time, should only the last one be sent and the other ones discarded?

Reply:
In the long run: yes. But for now, as discussed in chat, we can send all of the missing notifications, if you don't find an easy way to avoid it.

Reply:
I changed it so now it only sends the latest required notification of each process per run of the handler.